data/conf/nginx/includes/site-defaults.conf

```diff
@@ -7,26 +7,43 @@
+  #only supporting strong ciphers  
+  ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+  #prioritize ChaCha ciphers  
+  ssl_conf_command Options PrioritizeChaCha;
   ssl_ecdh_curve X25519:X448:secp384r1:secp256k1;
   ssl_session_cache shared:SSL:50m;
   ssl_session_timeout 1d;
   ssl_session_tickets off;
 
-  add_header Strict-Transport-Security "max-age=15768000;";
+  #includeSubDomains; and preload; if all of your services are using HTTPS
+  add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
   add_header X-Content-Type-Options nosniff;
-  add_header X-XSS-Protection "1; mode=block";
+  #X-XSS-Protection with Content Security instead
+  add_header X-XSS-Protection "0";
   add_header X-Robots-Tag none;
   add_header X-Download-Options noopen;
   add_header X-Frame-Options "SAMEORIGIN" always;
   add_header X-Permitted-Cross-Domain-Policies none;
   add_header Referrer-Policy strict-origin;
+  #Permission Policy  
+  add_header Permissions-Policy "accelerometer=(), ambient-light-sensor=(), autoplay=(), battery=(), browsing-topics=(), camera=(), clipboard-read=(), display-capture=(), document-domain=(), encrypted-media=(), fullscreen=(), geolocation=(), gyroscope=(), hid=(), idle-detection=(), interest-cohort=(), magnetometer=(), microphone=(), midi=(), payment=(), picture-in-picture=(), screen-wake-lock=(), serial=(), usb=(), sync-xhr=(), xr-spatial-tracking=()";
+  #If you use Gravatar with SOGo
+  add_header Content-Security-Policy "default-src 'none'; connect-src 'self' https://api.github.com https://www.gravatar.com; font-src 'self' https://fonts.gstatic.com; img-src 'self' data: https://www.gravatar.com; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com; frame-ancestors 'none'; upgrade-insecure-requests; block-all-mixed-content; base-uri 'none'";
+  #If you do not use Gravatar with SOGo
+  #add_header Content-Security-Policy "default-src 'none'; connect-src 'self' https://api.github.com; font-src 'self' https://fonts.gstatic.com; img-src 'self' data:; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com; frame-ancestors 'none'; upgrade-insecure-requests; block-all-mixed-content; base-uri 'none'";
+  #Cross-Origin Resource, Opener, and Embedder Policies
+  #add_header Cross-Origin-Resource-Policy same-origin;
+  #add_header Cross-Origin-Opener-Policy same-origin;
+  #if you do not use Gravatar with SOGo
+  #add_header Cross-Origin-Embedder-Policy require-corp;
 
   index index.php index.html;
 
   client_max_body_size 0;
 
-  gzip on;
+  #Disable gzip compression to avoid the BREACH attack
```

Comment:
Is there any source of this being actual in 2024?

Reply:
https://nginx.org/en/docs/http/ngx_http_gzip_module.html:
> When using the SSL/TLS protocol, compressed responses may be subject to [BREACH](https://en.wikipedia.org/wiki/BREACH) attacks.

---

### So it must be sth. else that mitigate that:
#### BREACH

The Browser Reconnaissance and Exfiltration via Adaptive Compression of Hypertext (BREACH) vulnerability is very similar to CRIME but BREACH targets HTTP compression, not TLS compression. This attack is possible even if TLS compression is turned off. An attacker forces the victim’s browser to connect to a TLS-enabled third-party website and monitors the traffic between the victim and the server using a man-in-the-middle attack. The BREACH vulnerability is registered in the NIST NVD database as [CVE-2013-3587](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2013-3587).

##### A vulnerable web application must satisfy the following conditions:
-   Be served from a server that uses HTTP-level compression
-   Reflect user input in HTTP response bodies
-   Reflect a secret (such as a CSRF token) in HTTP response bodies (therefore values in HTTP headers, such as cookies, are safe from this attack).

##### Prevention
-  Disable HTTP compression
-   Separate secrets from user input
-   Randomize secrets per request
-   Mask secrets (effectively randomize by XORing with a random secret per request)
-   Protect pages against CSRF
-   Hide the length (by adding random numbers of bytes to responses)
-   Limit the rate of requests

Source: https://www.acunetix.com/blog/articles/tls-vulnerabilities-attacks-final-part/

---

Maybe the CSRF Token in /web is enough to mitigate?
https://github.com/search?q=repo%3Amailcow%2Fmailcow-dockerized%20CSRF&type=code

But in my opinion the saver way is to turn gzip off as also bandwidth isn't the main reason anymore
so loading would increase just slightly? (haven't done any testing)